src/windows.rs

```diff
@@ -80,24 +82,36 @@ impl MmapInner {
         unsafe {
             let handle = kernel32::CreateFileMappingW(winapi::INVALID_HANDLE_VALUE,
                                                       ptr::null_mut(),
-                                                      prot.as_mapping_flag(),
+                                                      winapi::PAGE_EXECUTE_READWRITE,
                                                       (len >> 16 >> 16) as winapi::DWORD,
                                                       (len & 0xffffffff) as winapi::DWORD,
                                                       ptr::null());
             if handle == ptr::null_mut() {
                 return Err(io::Error::last_os_error());
             }
-            let ptr = kernel32::MapViewOfFile(handle, prot.as_view_flag(), 0, 0, len as winapi::SIZE_T);
+            let ptr = kernel32::MapViewOfFile(
+                handle,
+                winapi::FILE_MAP_ALL_ACCESS | winapi::FILE_MAP_EXECUTE,
+                0, 0, len as winapi::SIZE_T);
             kernel32::CloseHandle(handle);
 
             if ptr == ptr::null_mut() {
-                Err(io::Error::last_os_error())
-            } else {
+                return Err(io::Error::last_os_error());
+            }
+
+            let mut old = 0;
+            let result = kernel32::VirtualProtect(ptr,
```

Comment:
Could you add a comment about why the `VirtualProtect` is necessary here instead of handling it in `CreateFileMappingW`?  If it's explained in MSDN docs a link to that would work as well.